src/backends/torch/torchUtils.h

```diff
@@ -0,0 +1,47 @@
+#ifndef TORCH_UTILS_H
+#define TORCH_UTILS_H
+
+#if !defined(CPU_ONLY)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#include <c10/cuda/CUDACachingAllocator.h>
+#pragma GCC diagnostic pop
+#endif
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#include <torch/torch.h>
+#pragma GCC diagnostic pop
+#include <torch/script.h>
+
+namespace dd
+{
+
+  namespace torch_utils
+  {
+    inline void empty_cuda_cache()
+    {
+#if !defined(CPU_ONLY)
+      c10::cuda::CUDACachingAllocator::emptyCache();
```

Comment:
I know this is was not added by this PR, but beware of this, as my understanding is that it affects the global CUDA state. Do we know whether this is compatible with using multiple libtorch services ? cc @BynaryCobweb

Reply:
This method is thread safe ([code](https://github.com/pytorch/pytorch/blob/8357e2edc329b3a62979ceae3faa1e51fd70150e/c10/cuda/CUDACachingAllocator.cpp#L377)) and only free unused memory, so I think using it is not a problem.